pkg/operator/controller/dns_status.go

```diff
@@ -25,6 +29,11 @@ func (r *reconciler) syncDNSStatus(dns *operatorv1.DNS, clusterIP, clusterDomain
 	updated.Status.ClusterDomain = clusterDomain
 	updated.Status.Conditions = computeDNSStatusConditions(dns, clusterIP, haveDNSDaemonset, dnsDaemonset, haveNodeResolverDaemonset, nodeResolverDaemonset)
 	if !dnsStatusesEqual(updated.Status, dns.Status) {
+		// skip frequent updates to progressing and degraded conditions to reduce status flapping.
+		if conditionsUpdatedWithinLast(dns.Status.Conditions, conditionUpdateInterval) {
+			logrus.Infof("last condition update was less than %s ago, skipping this update.", conditionUpdateInterval)
```

Comment:
I'm wondering whether we want to spam the logs here.

Reply:
I changed it to debug.